manifests/bc.yaml

```diff
@@ -22,7 +22,54 @@ parameters:
     name: REPO_REF
     value: master
 objects:
-  - kind: "BuildConfig"
+  # use our own "gated" imagestream for the Jenkins master so we can test new
+  # images before upgrading
+  - apiVersion: v1
+    kind: ImageStream
```

Comment:
I think I'd feel better about this if we had the jenkins buildconfig in it's own separate yaml file. Any good reason to keep it in here other than reducing the comands to run by 1 ?

Reply:
looks like we are reusing the REPO_URL and REPO_REF below. so that might mean we can't break it out in a separate file

Reply:
Yeah exactly. The reason for keeping them together is so we can share template vars, as well as make set up slightly easier.

> I think I'd feel better about this if we had the jenkins buildconfig in it's own separate yaml file.

Anything in particular that you don't like? I'm fond of the "uni-template" approach as a way to have everything in one place. I think it's a pretty standard approach for OpenShift (see e.g. the Jenkins template we instantiate from for example: https://github.com/openshift/jenkins/blob/76b2cae92c0e5ae4837a83cd5402031d648d1412/openshift/templates/jenkins-persistent.json).

For legibility, note I added some dividers in #19.